File: Programmers/Lv0/진료순서정하기.cpp
#include <string>
#include <vector>
#include <map>
#include <algorithm>
using namespace std;

vector<int> solution(vector<int> emergency) {
    vector<int> answer;
    map<int, int> my_map;
    vector<int> sort_emergency = emergency;
    sort(sort_emergency.begin(), sort_emergency.end());
    for (int i = 0; i < sort_emergency.size(); i++)
        my_map[sort_emergency[i]] = emergency.size() - i;
    for (int i : emergency)
        answer.push_back(my_map[i]);
    return answer;
}

File: Programmers/Lv0/진료순서정하기.py
# 정렬을 통해 각 응급도에 따른 진료 순서를 딕셔너리 키, 값을 통해 배정한다.
# emergency배열의 값(응급도)에 해당하는 진료순서를 딕셔너리에서 뽑아 리스트에 넣는다.
def solution(emergency):
    answer = []
    dict = {}
    sort_emergency = sorted(emergency)
    for i in range(len(sort_emergency)):
        dict[sort_emergency[i]] = len(emergency) - i
    for i in emergency:
        answer.append(dict[i])
    return answer